tools.py

```diff
@@ -430,6 +430,41 @@ def processcomments(input_path, output_path, comment_path):
 	#print('\n'.join('{w},{be},{be_}'.format(w = w, be = not_word_stats[w]['be'], be_ = not_word_stats[w]['b'] + not_word_stats[w]['e']) for w in not_word))
 
 
+def filter_dataset(input_path,
+		output_path,
+		duration_in_hours,
+		cer,
+		seed):
+	dataset = json.load(open(input_path))
+
+	random.seed(seed)
+	random.shuffle(dataset)
+
+	print('initial set hours: ', sum(transcripts.compute_duration(t, hours=True) for t in dataset), 'hours')
+	if cer:
+		dataset = [e for e in dataset if e['cer'] <= cer]
+		print('after cer filtering hours: ', sum(transcripts.compute_duration(t, hours=True) for t in dataset), 'hours')
+
+	if duration_in_hours is not None:
+		s = []
+		set_duration = 0
+		while set_duration <= duration_in_hours and len(dataset) > 0:
+			t = dataset.pop()
+			set_duration += transcripts.compute_duration(t, hours = True)
+			s.append(t)
+		dataset = s
+
+	print('after duration filtering hours: ', sum(transcripts.compute_duration(t, hours=True) for t in dataset), 'hours')
+	print(output_path)
+	json.dump(
```

Comment:
please use `transcripts.save` instead of `json.dump`. we're slowly moving to the new API. with `transcripts.load`/`transcripts.save` it would be easier to migrate to other transcript formats